### ISSUE TITLE
Search flyout: there are focusable elements within an ARIA role that defines its children as presentational

### ISSUE THREAD

**Issue by msiewierska** [shopware/shopware#13972]
### Shopware Version

6.7.3.1

### Affected area / extension

Platform(Default)

### Actual behaviour

![Image](https://github.com/user-attachments/assets/437bbf22-6928-4301-8456-4edfeec1c03f)

WCAG 2.1 success criterion A 4.1.2 is violated with error: Aria-widget has an incorrect own element.

The problem is that the following element:
`<ul class="search-suggest-container" id="search-suggest-listbox" aria-label="Die Dropdown-Suche" role="listbox">`
 has the list of suggestion with focuable element inside:
`<li class="search-suggest-product js-result" role="option">
<a href="...." title="..." class="search-suggest-product-link">
....
`

### Expected behaviour

According to specification of ARIA: option role: 
_[All descendants are presentational](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/option_role#all_descendants_are_presentational)
There are some types of user interface components that, when represented in a platform accessibility API, can only contain text. Accessibility APIs do not have a way of representing semantic elements contained in an option. To deal with this limitation, browsers, automatically apply role [presentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/presentation_role) to all descendant elements of any option element as it is a role that does not support semantic children._

### How to reproduce

Insert search query inside search input in header. Search suggestions flyout is open.

**Comment by msiewierska** [comment]
Proper image from accessibility testing is:

<img width="1175" height="465" alt="Image" src="https://github.com/user-attachments/assets/c22b55ae-7126-4f6c-822a-8deb5e41257f" />
